cmd/common/image/inspect.go

```diff
@@ -127,12 +129,41 @@ func NewInspectCmd(runtime runtime.Runtime) *cobra.Command {
 			}
 			fmt.Fprint(cmd.OutOrStdout(), string(bytes))
 		case utils.OutputModeCustom:
+
 			extraInfo, _ := cmd.PersistentFlags().GetString("extra-info")
 			if extraInfo == "" {
 				return fmt.Errorf("extra info not specified (see --extra-info)")
 			}
-			if info.ExtraInfo.Data[extraInfo] == nil {
-				return fmt.Errorf("extra info %q not found", extraInfo)
+
+			jsonPath, _ := cmd.PersistentFlags().GetString("jsonpath")
+
+			if jsonPath != "" {
+				dataEntry, ok := info.ExtraInfo.Data[extraInfo]
+				if !ok {
+					return fmt.Errorf("extra info %q not found", extraInfo)
+				}
+				if dataEntry.ContentType != "application/json" {
+					return fmt.Errorf("extra info %q is not JSON", extraInfo)
+				}
+
+				// unmarshal the JSON content
+				var dataEntryContent []interface{}
```

Comment:
This works for json arrays but not for json maps.

With `--extra-info=oci.manifest` I get:
```
Error: unmarshalling JSON content: json: cannot unmarshal object into Go value of type []interface {}
```

I suggest to try with `[]interface{}` and if that does not work, to fallback on `map[string]interface{}`.